.circleci/config.yml

```diff
@@ -22,144 +21,164 @@ jobs:
   deploy:
     docker:
-      - image: harbor.k8s.libraries.psu.edu/library/ci-utils:$CI_UTILS_IMG_TAG
+      - image: harbor.k8s.libraries.psu.edu/library/ci-utils:$CI_UTILS_IMAGE_TAG
     environment:
       CONFIG_REPO: git@github.com:psu-libraries/scholarsphere-config.git
-      IMAGE_REPOSITORY: harbor.k8s.libraries.psu.edu/library/scholarsphere
-    parameters:
-      config_env:
-        type: string
     steps:
       - add_ssh_keys
       - run:
           name: "Updating Config Repo"
           command: |
-            export CONFIG_ENV=<< parameters.config_env >>
             ssh-keyscan github.com > ~/.ssh/known_hosts
             git clone $CONFIG_REPO
             cd scholarsphere-config
             ./generate_circle_application.sh
+  test:
+    docker:
+      - image: circleci/ruby:2.7.3-buster-node-browsers
+      - image: redis:6.2.6
+      - image: minio/minio:RELEASE.2021-03-17T02-33-02Z
+        entrypoint:
+        - /bin/sh
+        - -c
+        - mkdir -p /data/scholarsphere; minio --compat server --address ':9000' /data
+        environment:
+          MINIO_ACCESS_KEY: scholarsphere
+          MINIO_SECRET_KEY: scholarsphere
+      - image: postgres:11.8
+        environment:
+          POSTGRES_PASSWORD: scholarsphere
+          POSTGRES_USER: scholarsphere
+          POSTGRES_DB: scholarsphere
+      - image: solr:8.11.1-slim
+        command: [
+          "/bin/bash",
+          "-c",
+          "solr -c && solr auth enable -credentials scholarsphere:scholarsphere -z localhost:9983; solr stop && solr -c -f",
+        ]
+    environment:
+      RAILS_ENV: test
+      AWS_ACCESS_KEY_ID: scholarsphere
+      AWS_SECRET_ACCESS_KEY: scholarsphere
+      S3_ENDPOINT: http://localhost:9000
+      AWS_BUCKET: scholarsphere
+      AWS_REGION: us-east-1
+      SOLR_HOST: 127.0.0.1
+      SOLR_COLLECTION: scholarsphere
+      SOLR_PASSWORD: scholarsphere
+      SOLR_USERNAME: scholarsphere
+      POSTGRES_USERNAME: scholarsphere
+      POSTGRES_PASSWORD: scholarsphere
+      POSTGRES_DB: scholarsphere
+      POSTGRES_HOST: 127.0.0.1
+    steps:
+    - checkout
+    - restore_cache:
+        keys:
+          - bundle1-{{ checksum "Gemfile.lock" }}
+    - restore_cache:
+        keys:
+          - yarn1-{{ checksum "yarn.lock" }}
+    - run:
+        name: "Update Node.js and npm"
+        command: |
+          curl -sSL "https://nodejs.org/dist/v16.13.1/node-v16.13.1-linux-x64.tar.xz" | sudo tar --strip-components=2 -xJ -C /usr/local/bin/ node-v16.13.1-linux-x64/bin/node
+          curl https://npmjs.com/install.sh | sudo bash
+    - run:
+        name: "Bundle"
+        command: |
+          gem install bundler
+          bundle check --path vendor/bundle || bundle install --deployment
+    - run:
+        name: "Yarn"
+        command: |
+          yarn install --cache-folder ~/.cache/yarn --frozen-lockfile
+    # - run:
+    #     name: "Wait for Redis"
+    #     command: dockerize -wait tcp://localhost:6379 -timeout 1m
```

Comment:
Should this just be removed since it's commented out?

Reply:
good catch. i was testing if we needed  redis or not for the tests to run (i wasn't sure if they were mocked out in RAILS_ENV=test, i forgot to readd this.